client/service/src/builder.rs

```diff
@@ -170,27 +170,39 @@ impl KeystoreContainer {
 }
 
 /// Creates a new full client for the given config.
-pub fn new_full_client<TBl, TRtApi, TExec>(
+pub fn new_full_client<TBl, TRtApi, TExec, TBuildGenesisBlock>(
 	config: &Configuration,
 	telemetry: Option<TelemetryHandle>,
 	executor: TExec,
+	backend: Arc<TFullBackend<TBl>>,
+	genesis_block_builder: TBuildGenesisBlock,
 ) -> Result<TFullClient<TBl, TRtApi, TExec>, Error>
 where
 	TBl: BlockT,
 	TExec: CodeExecutor + RuntimeVersionOf + Clone,
+	TBuildGenesisBlock: BuildGenesisBlock<
+		TBl,
+		BlockImportOperation = <Backend<TBl> as sc_client_api::backend::Backend<TBl>>::BlockImportOperation
+	>,
 {
-	new_full_parts(config, telemetry, executor).map(|parts| parts.0)
+	new_full_parts(config, telemetry, executor, backend, genesis_block_builder).map(|parts| parts.0)
 }
 
 /// Create the initial parts of a full node.
-pub fn new_full_parts<TBl, TRtApi, TExec>(
+pub fn new_full_parts<TBl, TRtApi, TExec, TBuildGenesisBlock>(
```

Comment:
Please add asecond version of this function `new_full_parts_with_genesis_builder` or something like this. This variant here should use the default `GenesisBlockBuilder`.